### PR TITLE
📝Add End-to-End Documentation

### PR DESCRIPTION
**Pull request by dominikkaminski**
**Changes:**

1. Add documentation for E2E tests
2. Add logo of crossbrowsertesting.com
2. Add `processEngineUrl` env variable

## PR-Checklist

Please check the boxes in this list after submitting your PR:

- [x] You can merge this PR **right now** (if not, please prefix the title with "WIP: ")
- [x] I've tested **all** changes included in this PR.
- [x] I've also reviewed this PR myself before submitting (e.g. for scrambled letters, typos, etc.).
- [x] I've rebased the `develop` branch with my branch before finishing this PR.
- [x] I've **summarized all changes** in a list above.
- [x] I've mentioned all **PRs, which relate to this one**.
- [x] I've prefixed my Pull Request title is according to [gitmoji guide](https://gitmoji.carloscuesta.me/).

## Example

<details>
<summary>
:warning: Before merging please click here to expand and follow the guide.
</summary>
<br>

Please use `:twisted_rightwards_arrows:` at the beginning of your merge commit title.

Example 1:

<pre>
<code>
:twisted_rightwards_arrows: :bug: Fix Wrong Text Decoration at ...
</code>
</pre>

To get your commit message, just copy the first part of this pull request.

Example 2:
<pre>
<code>
**Changes:**

- Fixes Wrong Text Decoration at ...
- Fixes some typos
- ...

**Issues:**

Closes #NumberOfFixedIssue

PR: #NumberOfThisPR
</code>
</pre>
</details>

## Emojis

<details>
<summary>
Expand for a list of most used Emojis.
</summary>
<br>

Please prefix your commit messages with an Emoji.

Ref: https://gitmoji.carloscuesta.me/

| Description              | Glyphe               | Emoji  |
|--------------------------|----------------------|--------|
| Bugfix                   | `:bug:`              | 🐛     |
| Configuration releated   | `:wrench:`           | 🔧     |
| Cosmetic                 | `:lipstick:`         | 💄     |
| Dependencies Downgrade   | `:arrow_down:`       | ⬇️     |
| Dependencies Upgrade     | `:arrow_up:`         | ⬆️     |
| Formatting               | `:art:`              | 🎨     |
| Improving Performance    | `:zap:`              | ⚡️     |
| Initial commit           | `:tada:`             | 🎉     |
| Linter                   | `:rotating_light:`   | 🚨     |
| Miscellaneous            | `:package:`          | 📦     |
| New Feature              | `:sparkles:`         | ✨     |
| Refactoring Code         | `:recycle:`          | ♻️     |
| Releasing / Version tags | `:bookmark:`         | 🔖     |
| Removing Stuff           | `:fire:`             | 🔥     |
| Tests                    | `:white_check_mark:` | ✅     |
| Work In Progress (WIP)   | `:construction:`     | 🚧     |

</details>

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -185,26 +185,106 @@ docker run -p 17290:17290 bpmn:v0.1
 
 Anschließend lässt sich das BPMN-Studio unter URL `http://localhost:17290` aufrufen.
 
-### End-to-End-Tests
+## End-to-End-Tests
 
-Start des Webservers:
+### Vorraussetzungen
```
Comment (Vyperus) [inline README.md]: Niemals eine Überschrift auf eine andere folgen lassen.
Vorschlag: 

## End-to-End-Tests

(Kurzer Text darüber was und warum)

### Nächste Überschrift

File: README.md [hunk 2]
```diff
@@ -185,26 +185,106 @@ docker run -p 17290:17290 bpmn:v0.1
...
+* Docker
+
+Nur bei CrossBrowserTesting:
+* VM oder vergleichbares mit öffentlicher IP
+* Docker
+* CrossBrowserTesting account
+
+### Lokale tests ohne Docker
+
+Initial müssen folgende Befehle in einem Terminal ausgeführt werden:
```
Comment (Vyperus) [inline README.md]: Ein kurzer Satz darüber das alle folgenden Punkte im neu geklonten BPMN-Studio Ordner erfolgen müssen wäre imo nicht schlecht.

File: README.md [hunk 3]
```diff
@@ -185,26 +185,106 @@ docker run -p 17290:17290 bpmn:v0.1
...
+npm run build
+```
+
+Der Webservice kann mit folgendem Befehl gestartet werden:
+
+```shell
+npm start -- --port=9000
+```
+
+Ein weiteres Terminal öffnen und folgenden Befehl eingeben um die Process Engine Runtime zu starten:
```
Comment (robpal94) [inline README.md]: Funktioniert das aktuell mit der ProcessEngine Runtime? @cmg-dev sagte mir, dass es damit noch eventuell Probleme geben könnte.

File: README.md (hunk 2 above)
Comment (dominikkaminski) [reply README.md]: Done ✔

File: README.md (hunk 3 above)
Comment (dominikkaminski) [reply README.md]: Habe ich ebenfalls hinterlegt.

**Comment by dominikkaminski** [comment]
@ElRaptorus @Vyperus Please recheck 🙏

File: README.md [hunk 4]
```diff
@@ -185,26 +185,112 @@ docker run -p 17290:17290 bpmn:v0.1
 
 Anschließend lässt sich das BPMN-Studio unter URL `http://localhost:17290` aufrufen.
 
-### End-to-End-Tests
+## End-to-End-Tests
 
-Start des Webservers:
+Die End-to-End-Tests werden mit Hilfe des [Protractor Frameworks](https://www.protractortest.org/#/) durchgeführt.
+
+**Hinweis:** Alle Befehle müssen inm geklonten Repository ausgeführt werden.
```
Comment (robpal94) [inline README.md]: Typo: _müssen **im geklontem** Repository ausgeführt werden..._

File: README.md [hunk 5]
```diff
@@ -185,26 +185,112 @@ docker run -p 17290:17290 bpmn:v0.1
...
+
+**Hinweis:** Aktuell schlagen einige Test fehl aufgrund eines Bugs in der Process Engine Runtime [#9](https://github.com/process-engine/process_engine_runtime/issues/9)
```
Comment (robpal94) [inline README.md]: _Aktuell schlagen aufgrund eines Fehlers innerhalb der `ProcessEngine Runtime` fehl._

File: README.md [hunk 6]
```diff
@@ -185,26 +185,112 @@ docker run -p 17290:17290 bpmn:v0.1
...
 
-### End-to-End-Tests
+## End-to-End-Tests
 
-Start des Webservers:
+Die End-to-End-Tests werden mit Hilfe des [Protractor Frameworks](https://www.protractortest.org/#/) durchgeführt.
+
+**Hinweis:** Alle Befehle müssen im geklonten Repository ausgeführt werden.
+
+**Hinweis:** Aktuell schlagen einige Test fehl aufgrund eines Bugs in der Process Engine Runtime [#9](https://github.com/process-engine/process_engine_runtime/issues/9)
```
Comment (alexanderkasten) [inline README.md]: Tests

Comment (alexanderkasten) [reply README.md]: ProcessEngine

File: README.md [hunk 7]
```diff
@@ -185,26 +185,112 @@ docker run -p 17290:17290 bpmn:v0.1
...
+
+### Vorraussetzungen
+
+Es werden folgende Pakete benötigt:
+
+Allgemein:
+* [Java Development Kit](http://www.oracle.com/technetwork/java/javase/downloads/jdk8-downloads-2133151.html)
+* [Chrome](https://www.google.com/intl/de_ALL/chrome/)
+
+Nur bei lokalen tests:
```
Comment (alexanderkasten) [inline README.md]: Tests

File: README.md [hunk 8]
```diff
@@ -185,26 +185,112 @@ docker run -p 17290:17290 bpmn:v0.1
...
+* [Process Engine](https://www.npmjs.com/package/@process-engine/process_engine_runtime)
+
+Nur bei lokalen tests mit Docker:
```
Comment (alexanderkasten) [inline README.md]: Tests

File: README.md [hunk 9]
```diff
@@ -185,26 +185,112 @@ docker run -p 17290:17290 bpmn:v0.1
...
+* Docker
+
+Nur bei CrossBrowserTesting:
+* VM oder vergleichbares mit öffentlicher IP
+* Docker
+* CrossBrowserTesting account
```
Comment (alexanderkasten) [inline README.md]: Account

File: README.md [hunk 10]
```diff
@@ -185,26 +185,112 @@ docker run -p 17290:17290 bpmn:v0.1
...
 ```
 
-Ein anderes Terminal öffnen und den Selenium Server starten:
+Der Webservice kann mit folgendem Befehl gestartet werden:
+
+```shell
+npm start -- --port=9000
+```
+
+Ein weiteres Terminal öffnen und folgenden Befehl eingeben um die Process Engine Runtime zu starten:
```
Comment (alexanderkasten) [inline README.md]: , um die ProcessEngine

File: README.md [hunk 11]
```diff
@@ -185,26 +185,112 @@ docker run -p 17290:17290 bpmn:v0.1
...
 ```
 
-Die Tests starten:
+Die End-to-End tests können in einem weiteren Terminal mit folgendem Befehl gestartet werden:
 
 ```shell
 npm run integration-test
 ```
 
+### Lokale tests mit Docker
```
Comment (alexanderkasten) [inline README.md]: Tests

File: README.md [hunk 12]
```diff
@@ -185,26 +185,112 @@ docker run -p 17290:17290 bpmn:v0.1
...
+
+Ein Terminal öffnen und das BPMN Studio mit folgendem Befehl starten:
```
Comment (alexanderkasten) [inline README.md]: BPMN-Studio

File: README.md [hunk 13]
```diff
@@ -185,26 +185,112 @@ docker run -p 17290:17290 bpmn:v0.1
...
+
+```shell
+docker run -p 8000:8000 -p 9000:9000 5minds/bpmn-studio-bundle:latest
+```
+
+Ein weiteres Terminal öffnen und folgenden Befehl eingeben um den Selenium Server zu starten:
```
Comment (alexanderkasten) [inline README.md]: , um

File: README.md [hunk 14]
```diff
@@ -185,26 +185,112 @@ docker run -p 17290:17290 bpmn:v0.1
...
+
+```shell
+npm run integration-test-init
+```
+
+Die End-to-End tests können in einem weiteren Terminal mit folgendem Befehl gestartet werden:
```
Comment (alexanderkasten) [inline README.md]: Tests

File: README.md [hunk 15]
```diff
@@ -185,26 +185,112 @@ docker run -p 17290:17290 bpmn:v0.1
...
+
+```shell
+npm run integration-test
+```
+
+### Crossbrowser tests mit Docker
```
Comment (alexanderkasten) [inline README.md]: Tests

File: README.md [hunk 16]
```diff
@@ -185,26 +185,112 @@ docker run -p 17290:17290 bpmn:v0.1
...
+[<img src="https://crossbrowsertesting.com/design/images/brand/cbt-sb-logo.svg" width="250px">](https://crossbrowsertesting.com)
+
+Initial müssen folgende Environment Variablen gesetzt werden:
```
Comment (alexanderkasten) [inline README.md]: Umgebungsvariablen

File: README.md [hunk 17]
```diff
@@ -185,26 +185,112 @@ docker run -p 17290:17290 bpmn:v0.1
...
+
+```shell
+export CB_USER=""           # CrossBrowserTesting E-Mail Adresse
+export CB_KEY=""            # CrossBrowserTesting API Key
+export aureliaUrl=""        # URL oder IP der VM + Port des BPMN-Studio; z.B.: http://1.1.1.1:9000
+export processEngineUrl=""  # URL oder IP der VM + Port der Process Engine; z.B.: http://1.1.1.1:8000
```
Comment (alexanderkasten) [inline README.md]: ProcessEngine

File: README.md [hunk 18]
```diff
@@ -185,26 +185,112 @@ docker run -p 17290:17290 bpmn:v0.1
...
+```
+
+Ein Terminal öffnen und das BPMN Studio auf der VM mit folgendem Befehl starten:
```
Comment (alexanderkasten) [inline README.md]: BPMN-Studio

File: README.md [hunk 19]
```diff
@@ -185,26 +185,112 @@ docker run -p 17290:17290 bpmn:v0.1
...
+
+```shell
+docker run -p 8000:8000 -p 9000:9000 5minds/bpmn-studio-bundle:latest
+```
+
+Ein Terminal auf dem lokalen Computer öffnen und mit folgenden Befehl die CrossBrowserTesting tests starten:
```
Comment (alexanderkasten) [inline README.md]: Tests

File: README.md [hunk 20]
```diff
@@ -185,26 +185,112 @@ docker run -p 17290:17290 bpmn:v0.1
...
 
-### End-to-End-Tests
+## End-to-End-Tests
 
-Start des Webservers:
+Die End-to-End-Tests werden mit Hilfe des [Protractor Frameworks](https://www.protractortest.org/#/) durchgeführt.
+
+**Hinweis:** Alle Befehle müssen im geklonten Repository ausgeführt werden.
+
+**Hinweis:** Aktuell schlagen einige Tests fehl aufgrund eines Bugs in der ProcessEngine [#9](https://github.com/process-engine/process_engine_runtime/issues/9)
```
Comment (robpal94) [inline README.md]: _Aufgrund eines Fehlers innerhalb der `ProcessEngine Runtime`, schlagen aktuell  einige Tests fehl._

Comment (Vyperus) [reply README.md]: @dominikkaminski Bitte noch umsetzen

File: integration-test.sh [hunk 21]
```diff
@@ -3,5 +3,6 @@
 export CB_USER=""
 export CB_KEY=""
 export aureliaUrl=""
+export processEngineUrl="" 
 
 # Run this command to apply environment variables: source integration-test.sh
```
Comment (cmg-dev) [inline integration-test.sh]: 1. Bitte verschieben

  in den Ordner: `bpmn-studio/dev_tooling/scripts/`

1. und besser benennen: 

   `prepare_environment_for_integration_tests.sh`

Tipp:

Generell immer Dinge so benennen, das klar ist, was sie tun!

**Pull request merged**